channels/sessions.py

```diff
@@ -221,12 +220,15 @@ async def send(self, message):
         # Pass up the send
         return await self.real_send(message)
 
-    def save_session(self):
+    async def save_session(self):
         """
         Saves the current session.
         """
         try:
-            self.scope["session"].save()
+            if django.VERSION >= (5, 1):
+                await self.scope["session"].asave()
+            else:
+                await database_sync_to_async(self.scope["session"].save)()
         except UpdateError:
             raise SuspiciousOperation(
                 "The request's session was deleted before the "
```

Comment:
I'm not actually sure how to trigger this logic, because based on reading the Django source code I can't figure out where it would be raised when calling `save()` without `must_create=True`

Reply:
Perhaps: 

* Create a session normally. 
* Await an `Event`. 
* Delete the session whilst waiting. (Simulating the logout)
* Try to save the session. (Should raise the `UpdateError`, I think 😅)

Reply:
Ah thank you! I managed to get it written after a bit of pain